inspirehep/modules/authors/views.py

```diff
@@ -225,26 +227,35 @@ def new():
     return render_template('authors/forms/new_form.html', form=form, **ctx)
 
 
+def _author_record_from_db(recid):
+    """Get author record from DB or return None if it is not there."""
+    try:
+        return get_db_record('aut', recid)
+    except RecordGetterError:
+        return None
+
+
+def _author_record_from_legacy(recid):
+    """Get author record from Legacy or return None if it is not there."""
+    try:
+        url = get_legacy_url_for_recid(recid) + '/export/xm'
+        xml_response = requests.get(url)
+        xml = fromstring(xml_response.content)
+        xml_record = xml.xpath("//*[local-name() = 'record']")[0]
+        return marcxml2record(tostring(xml_record))
+    except requests.exceptions.RequestException:
+        return None
+
+
 @blueprint.route('/<int:recid>/update', methods=['GET'])
 @register_breadcrumb(blueprint, '.update', 'Update author information')
 @login_required
 def update(recid):
     """View for INSPIRE author update form."""
-    data = {}
-    if recid:
-        try:
-            url = get_legacy_url_for_recid(recid) + '/export/xm'
-            xml = requests.get(url)
-            record_regex = re.compile(
-                r"\<record\>.*\<\/record\>", re.MULTILINE + re.DOTALL)
-            xml_content = record_regex.search(xml.content).group()
-            data = marcxml2record(xml_content)
-            convert_for_form(data)
-        except requests.exceptions.RequestException:
-            pass
-        data["control_number"] = recid
-    else:
-        return redirect(url_for("inspirehep_authors.new"))
+    data = _author_record_from_db(recid) or _author_record_from_legacy(recid) or {}
+    convert_for_form(data)
+    data["control_number"] = recid
```

Comment:
Wont you get a keyerror if the data is an empty object?

Reply:
```
In [4]: a_dict = {}

In [5]: a_dict['a_key'] = 'a_value'

In [6]: a_dict
Out[6]: {'a_key': 'a_value'}
```

Reply:
Ah, my bad didn't notice you're actually setting it :)